LearningMissionSimulation/PlaygroundGarush.cs

```diff
@@ -86,66 +116,170 @@ List<Module> GetModuleList()
                 if (!moduleList.Contains(module))
```

Comment:
You have to check the return value of moduleDictionary.TryGetValue(id, out module) -- before adding the module to the list.